Update better-gherkin.md

atdd-bdd opened the pull request:
Added an intermediate style between imperative and declarative

### 🤔 What's changed?

Added a third style between imperative and declarative

### ⚡️ What's your motivation? 

It's good to have three perspectives on how to approach an issue.   In many cases, the data in the scenario should be transparent.   The third way allows for transparency, as well as reuse of the scenario in multiple contexts.   

### 📋 Checklist:

- [x] I agree to respect and uphold the [Cucumber Community Code of Conduct](https://cucumber.io/conduct/)

atdd-bdd committed:
Update better-gherkin.md

Added an intermediate style between imperative and declarative

luke-hill commented on content/docs/bdd/better-gherkin.md:
```diff
@@ -80,3 +84,39 @@ Scenario: Subscriber with a paid subscription can access both free and paid arti
+
+## A third style 
+
+There is a style that is intermediate between these two.   The exact values are specified in the scenario, but the way the user interacts with the system is not.  This style uses stepdef tables with domain terms as the column headers.   The step definitions can  be reused for mulitple scenarios with different data in the table.  In this example, the scenarios for free and paid subscribers use the ...
```
Few minor things to fix:
- Check your formatter here as lots of extra spaces are being added.
- Use `step definitions` instead of stepdef
- `mulitple`* typo

Also these technical items are known as data tables not step definition tables

atdd-bdd replied:
Made the changes as you suggested.    Thanks for the corrections.

luke-hill commented on content/docs/bdd/better-gherkin.md:
```diff
@@ -80,3 +84,39 @@ Scenario: Subscriber with a paid subscription can access both free and paid arti
+| Title           | For Subscription  |
+| Free Article 1  | Free              |
+| Paid Article 1  | Paid              |
+And user is logged in 
```
if you're using a single table, maybe suffix it with `as` i.e. `And the user is logged in as` for both scenarios

atdd-bdd replied:
Made the change

luke-hill commented on content/docs/bdd/better-gherkin.md:
```diff
@@ -80,3 +84,39 @@ Scenario: Subscriber with a paid subscription can access both free and paid arti
+And user is logged in 
+| User Name               | Password          | Subscription  |
+| paidPattya@example.com  | validPassword123  | Paid          |
+When articles are displayed 
```
This step would then become redundant in both scenarios

atdd-bdd replied:
I was split on whether the login should become a separate scenario or not.   I was matching the data in the imperative style

There would at least need to be a step that said the logged user had a paid.free subscription.  Keeping that as a data table emphasizes that Subscription is a domain term.    

I could write a section on splitting scenarios, which might come after this one.

luke-hill reviewed (commented):
Check your editor as it seems as though some inconsistent spacing is being added.

atdd-bdd committed:
Update better-gherkin.md

atdd-bdd committed:
Update better-gherkin.md

luke-hill reviewed (commented):
Most of the sentences are good but it's likely you're missing the conjunction "the" in most/all of them (Possibly confusing if you're not a native speaker).

Also if you can remove the redundant steps we can get this merged.

atdd-bdd committed:
Update better-gherkin.md

Split off the login scenario from the display scenarios.   The logon follows along with the first scenario on this page.

atdd-bdd committed:
Update better-gherkin.md

Add conjunctions to the scenarios.

atdd-bdd commented:
I think this should meet what you asked for.   Let me know if I am still missing something.    Thanks.

atdd-bdd committed:
Merge branch 'cucumber:main' into patch-4

luke-hill commented:
Hi @atdd-bdd Think there's still a couple of things to fix up. Missed your comment sorry
